Rn2304 (#1187)
elasticio/elasticio.github.io: content/_releases/23/04.md

For more informaition on Embedded recipe deployment please read this [article](/guides/embedded-recipe-deployment/).

### Create/Edit Recipe API-Call

We are introdusing `POST /v2/flows/export-to-recipe` API Call. With this cola you can:

* export a set of flows to a recipe.
* update existing recipe with set of new flows.

This new, more powerful endpoint, replaces the old one - `POST /v2/flows/${flowId}/export-to-recipe`. Please visit [API Docs](({{site.data.tenant.apiDocsUri}}/v2#/recipes/post_flows_export_to_recipe)) for more details.

### Flow Queues improvements

Before, the Flow Queues page displays the components inverted according to the flow graph (e.g if graph contain fork or content-based router) and don't display the step name, only the number.

Now we render step name on Flow Queues page (like on Flow Designer) and render flow graph at Flow Queues, Runlog Execution, Thread pages in the same way as at Flow Designer page.

{% include img.html max-width="60%" url="/assets/img/RN/2252/rn2304-flow-queues.png" title="Flow Queues" %}

## Fixed bugs

> *   **FIXED** bug, where it failed to render flow-queue page after quick publishing a flow-draft and running or rendering the old flow graph if the flow has a previous version.

Reviewer: ```suggestion
*   **FIXED** bug, where the flow-queues page would not load after the draft publishing in case the flow had an earlier version.
```

---
title: Product Update - v23.04
description: Details of the product update v23.04 from 26th of January 2023.
layout: rel
version: "23.04"
section: Product Updates
category: product-updates
releaseDate: 2023-01-26
redirect_from:
  - /releases/23.04.html
---

## Improvements

### Embedded Recipe Page: Recipe Credentials Edit

> We are introducing the ability to create embedded page for editing credentials for a single resipe's deployment. For this purpose we have created a new API endpoint `GET /v2/recipe-deployments/:recipe_deployment_id`. This endpoint requires same permissions and scopes as in `/v2/recipe/:recipe_id?include=recipe-deployments`.

Reviewer 2: ```suggestion
We previously introduced the ability to add new credentials for a deployed recipe via an embedded page in a 3rd-party site or app. Today we introduce the ability to create an embedded page for editing credentials for a single recipe deployment. For this purpose we have created a new API endpoint `GET /v2/recipe-deployments/:recipe_deployment_id`. This endpoint requires same permissions and scopes as in `/v2/recipe/:recipe_id?include=recipe-deployments`.
```

> For more informaition on Embedded recipe deployment please read this [article](/guides/embedded-recipe-deployment/).

### Create/Edit Recipe API-Call

We are introdusing `POST /v2/flows/export-to-recipe` API Call. With this call you can:

* export a set of flows to a recipe.
* update existing recipe with set of new flows.

This new, more powerful endpoint, replaces the old one - `POST /v2/flows/${flowId}/export-to-recipe`. Please visit [API Docs](({{site.data.tenant.apiDocsUri}}/v2#/recipes/post_flows_export_to_recipe)) for more details.

### Flow Queues improvements

Before, the Flow Queues page displays the components inverted according to the flow graph (e.g if graph contain fork or content-based router) and don't display the step name, only the number.

Now we render step name on Flow Queues page (like on Flow Designer) and render flow graph at Flow Queues, Runlog Execution, Thread pages in the same way as at Flow Designer page.

{% include img.html max-width="60%" url="/assets/img/RN/2304/rn2304-flow-queues.png" title="Flow Queues" %}

{% if site.data.tenant.name == "elastic.io" %}

> 

Reviewer 2: ### Maximum frequency of flow scheduling changed to 10 minutes

As part of our work to make the platform more robust and tolerant of high loads, we have decreased the maximum frequency of scheduling a flow to start. This default setting ensures that flows will be started more precisely. Even if the flow schedule is defined in the settings page as running every 3 minutes, these flows will now run only every 10 minutes.
Customers wishing to trigger flows more frequently must change the flow to run in Real-time mode.

## OEM Related

### OEM Fixed bugs

> *   **FIXED** bug, where TCP connection spamming from API to Elasticsearch during mass log retrieval. For this purpose we have created a singleton Elasticsearch client that will be used by `v2/logs`

Reviewer 2: ```suggestion
*   **FIXED** bug, where the TCP connections were inefficiently making requests from our API to Elasticsearch during mass log retrieval. For this purpose we have created a singleton Elasticsearch client that will be used by `v2/logs`
```

{% endif %}

## Fixed bugs

> *   **FIXED** bug, where the flow-queues page would not load after the draft publishing in case the flow had an earlier version.

Reviewer 2: ```suggestion
*   **FIXED** bug, where the flow-queues page would not load directly after a draft-version of a previously published flow was published.
```

> *   **FIXED** bug, when the Credentials and **Recipe** page would not load in case when the user would not have permission to access secrets. Now the UI shows a message with an error.

Reviewer 2: ```suggestion
*   **FIXED** bug, when the Credentials and **Recipe** page would not load in cases where the user did not have permission to access auth secrets. Now, the UI shows a message with an error.
```
